ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoSpec.scala

```diff
@@ -1103,6 +1103,57 @@ class JdbcLedgerDaoSpec
         res3 shouldBe a[LedgerEntry.Rejection]
       }
     }
+
+    "be able to use divulged contract in later transaction" in {
+      val let = Instant.now
+      val offset1 = nextOffset()
+      val offset2 = nextOffset()
+      val offset3 = nextOffset()
+      def emptyTxWithDivulgedContracts(id: Long) = PersistenceEntry.Transaction(
+        LedgerEntry.Transaction(
+          Some(s"commandId$id"),
+          s"transactionId$id",
+          Some("applicationId"),
+          Some(alice),
+          Some("workflowId"),
+          let,
+          let,
+          GenTransaction(HashMap.empty, ImmArray.empty, None),
+          Map.empty
+        ),
+        Map.empty,
+        Map(AbsoluteContractId(s"contractId$id") -> Set(bob)),
+        List(AbsoluteContractId(s"contractId$id") -> someContractInstance)
+      )
+
+      for {
+        // First try and index a transaction fetching a completely unknown contract.
+        _ <- ledgerDao.storeLedgerEntry(offset1, offset1 + 1, None, txFetch(let, offset1, bob, 0))
+        res1 <- ledgerDao.lookupLedgerEntryAssert(offset1)
+
+        // Then index a transaction that just divulges the contract to bob.
+        _ <- ledgerDao.storeLedgerEntry(
+          offset2,
+          offset2 + 1,
+          None,
+          emptyTxWithDivulgedContracts(offset2))
+        res2 <- ledgerDao.lookupLedgerEntryAssert(offset2)
+
+        // Finally try and fetch the divulged contract. LedgerDao should be able to look up the divulged contract
+        // and index the transaction without finding the contract metadata (LET) for it, as long as the contract
+        // exists in contract_data.
+        _ <- ledgerDao.storeLedgerEntry(
+          offset3,
+          offset3 + 1,
+          None,
+          txFetch(let.plusSeconds(1), offset3, bob, offset2))
+        res3 <- ledgerDao.lookupLedgerEntryAssert(offset3)
+      } yield {
+        res1 shouldBe a[LedgerEntry.Rejection]
+        res2 shouldBe a[LedgerEntry.Transaction]
+        res3 shouldBe a[LedgerEntry.Transaction]
```

Comment:
Without your fix the last assertion fails, right?

Reply:
Yep.

Reply:
OK